Gordon360/Services/HousingService.cs

```diff
@@ -321,234 +326,217 @@ public int EditApplication(string editorId, string sess_cde, int apartAppId, str
-            {
-                foreach (ApartmentChoiceSaveViewModel apartmentChoiceModel in apartmentChoicesFromDB)
-                {
-                    string oldApartmentChoice = apartmentChoiceModel.BLDG_CDE;
-                    existingApartmentChoices.Add(oldApartmentChoice);
-                }
-            }
-
-            // Add the hall preference from the frontend into the list of strings
-            if (newApartmentChoices.Any())
-            {
-                foreach (ApartmentChoiceViewModel newApartmentChoiceModel in newApartmentChoices)
-                {
-                    string newApartmentChoice = newApartmentChoiceModel.HallName;
-                    newApartmentChoicesToAddOrUpdate.Add(newApartmentChoice);
-                }
-            }
-
-            // Find a list of apartment choices to remove
-            foreach (string bldg in existingApartmentChoices)
-            {
-                if (!newApartmentChoicesToAddOrUpdate.Contains(bldg))
-                {
-                    apartmentChoicesToRemove.Add(bldg);
-                    newApartmentChoicesToAddOrUpdate.Remove(bldg);
-                }
-            }
-
             // List of apartment choices that are in the array recieved from the frontend but not yet in the database
-            List<string> apartmentChoicesToAdd = new List<string>();
+            List<ApartmentChoiceViewModel> apartmentChoicesToAdd = new List<ApartmentChoiceViewModel>(newApartmentChoices);
+
             // List of apartment choices that are in both the array recieved from the frontend and the database
-            List<string> apartmentChoicesToUpdate = new List<string>();
+            List<ApartmentChoiceViewModel> apartmentChoicesToUpdate = new List<ApartmentChoiceViewModel>();
 
-            // Separate the list into two: ToAdd and ToUpdate
-            foreach (string bldg in newApartmentChoicesToAddOrUpdate)
+            // List of apartment choices that are in the database but not in the array recieved from the frontend
+            List<ApartmentChoiceViewModel> apartmentChoicesToRemove = new List<ApartmentChoiceViewModel>();
+
+            // Check whether any apartment choices were found matching the given application ID number
+            if (existingApartmentChoiceResult.Any())
             {
-                if (existingApartmentChoices.Contains(bldg))
-                {
-                    apartmentChoicesToUpdate.Add(bldg);
-                    newApartmentChoicesToAddOrUpdate.Remove(bldg);
-                }
-                else
+                foreach (GET_AA_APARTMENT_CHOICES_BY_APP_ID_Result existingApartmentChoice in existingApartmentChoiceResult)
                 {
-                    apartmentChoicesToAdd.Add(bldg);
+                    ApartmentChoiceViewModel newMatchingApartmentChoice = null;
+                    newMatchingApartmentChoice = newApartmentChoices.FirstOrDefault(x => x.HallName == existingApartmentChoice.BLDG_CDE);
```

Comment:
This code is very similar to the Applicants update code- it would be great if we could make a function out of the commonalities, but not necessary right now

Reply:
You can't do that in practice because of how the properties behave and having different structures in the view models.

In the future, this code will actually diverge more because the applicants and apartment choices will have more different things that need to be compared and handled differently, so it will best for the upcoming features if we leave it the way it is

Reply:
Sounds good